picard/ui/options/plugins.py

```diff
@@ -235,6 +235,16 @@ def __init__(self, parent=None):
         self.ui.folder_open.clicked.connect(self.open_plugin_dir)
         self.ui.reload_list_of_plugins.clicked.connect(self.reload_list_of_plugins)
 
+        # Keep track whether the object has been destroyed to avoid trying to update
+        # deleted UI widgets after plugin list refresh.
+        self._deleted = False
+
+        # The on destroyed cannot be created as a method on this class or it will never get called.
+        # See https://stackoverflow.com/questions/16842955/widgets-destroyed-signal-is-not-fired-pyqt
+        def on_destroyed(obj=None):
+            self._deleted = True
+        self.destroyed.connect(on_destroyed)
+
```

Comment:
I wonder if it wouldn't make sense to have this at `OptionsPage` class level.
After all, this issue isn't specific to plugins options pages, it has impact on it because it has running queries at the time it is closed, but this could be the case for any other options pages. In this case, `_delete` could be renamed `delete` and marked as property.

Reply:
You are right in general. But I checked the code and currently this is the only place where we call some threaded background code with a callback. And as I think this is kind of ugly I would prefer to contain this to the place where it is needed unless we need it somewhere else as well.

One argument against this would be if we want to make this available to plugins. But this would mean making it a somewhat official API. 

So not sure really. If you think it's better to have it centralized I can live with that as well :)

Reply:
Well, we can move it at any time, though making it available for any options pages (including those from plugins) would help in similar case (where callback's calls are delayed and page closed), so that's not purely a plugins options page thing. It seems more logical to me it is a feature of the parent class.
Even though I don't know any other options pages that might be concerned by this issue, we are pretty sure some plugins are actually doing queries using web service, and those are very likely to be concerned by this issue.

Reply:
Ok, convinced :) e29315c2fd61ad140f798eb57ce741c81e189b7c